saas/renewals.py

```diff
@@ -116,11 +116,26 @@ def _recognize_subscription_income(subscription, until=None):
                         'nb_periods': nb_periods,
                         'period_start': descr_period_start,
                         'period_end': descr_period_end})
+
             recognize_period_idx += 1
             recognize_start = (subscription.created_at
                 + relativedelta(months=recognize_period_idx))
             recognize_end = (subscription.created_at
                 + relativedelta(months=recognize_period_idx + 1))
+        use_charges = subscription.plan.use_charges.all()
+        use_charge_amount = 0
+        for use_charge in use_charges:
+            quantity = get_period_usage(subscription, use_charge,
```

Comment:
If `get_period_usage` returns both the `quantity` and `amount`, you don't need to recompute it here. To avoid discrepancies, it always best to compute balances out of the `Transaction` ledger (filter on the specific event_id).

Reply:
what do you mean by "out of the `Transaction` ledger"?

Reply:
What you have done is recompute the over-quota balance amount in code while that amount should be coming out of a SQL query - assuming `record_use_charge` is implemented to as previously described.

Reply:
`get_period_usage` returns a `count()` of transaction with corresponding `event_id`. Should we use `sum_dest_amount` or similar instead?